refractr/schema.yml

```diff
@@ -20,37 +20,48 @@ defs:
     title: if-predicate
     type: string
     pattern: '^(^|$).*$'
-  url:
+  test-url:
     default: ''
-    title: url
+    title: test-url
     type: string
-    pattern: '^https?://[^\s/.?#].[^\s]*$'
-  urls:
+    pattern: '^(https?:\/\/)([A-z0-9]+([A-z0-9\.\-]+)\/)([A-z0-9\-_]+\/)*([A-z0-9$\/\.\-_:]+)?(\?[A-z0-9\.\-:=]+(&[A-z0-9\.\-:=]+)*)?(#[A-z0-9\-_]+)?$'
```

Comment:
Is using a library that does parsing instead of a regex perhaps better here?

Reply:
Oh I wish, but that is not how this works to my understanding.  If you are using patternProperties, which we are, then you are writing your own regexes.  This is yaml that gets loaded, so it is text not code objects.